src/classes/STG_InstallScript.cls

```diff
@@ -337,25 +338,62 @@ global without sharing class STG_InstallScript implements InstallHandler {
             UTIL_OrgTelemetry.dispatchTelemetryBatchJob();
         }
     }
-    
-    /**********************************************************************************************************
-    * @description Enables prevent deletion by default on Hierarchy Custom Settings for new installs and 
-    * when upgrading to 1.95
+
+    /*********************************************************************************************************
+    * @description Enables Prevent Hierarchy Custom Settings for new installs and push upgrades
     * @return void
     */
     global static void enablePreventDeletionHierarchyCustomSettings(Version previousVersion) {
-        if ((previousVersion == NULL) || (previousVersion != null && previousVersion.compareTo(new Version(1,95)) < 0)) {
+        Hierarchy_Settings__c orgSettings = UTIL_CustomSettingsFacade.getSettings();
+        Boolean isNewInstallOrPushUpgradedTo195 = newInstallOrPushUpgrade(previousVersion, new Version(1,95));
+        Boolean isNewInstallOrPushUpgradedTo196 = newInstallOrPushUpgrade(previousVersion, new Version(1,96));
+        if (isNewInstallOrPushUpgradedTo195) {
             UTIL_Debug.debug('***Upgrading to version 1.95 or new installs');
-            Hierarchy_Settings__c orgSettings = UTIL_CustomSettingsFacade.getSettings();
-            
-            if (orgSettings.Prevent_Account_Deletion__c == null || orgSettings.Prevent_Account_Deletion__c != true) {
-                orgSettings.Prevent_Account_Deletion__c = true; 
-            }
-            if (orgSettings.Prevent_Plan_Requirement_Deletion__c == null || orgSettings.Prevent_Plan_Requirement_Deletion__c != true) {
-                orgSettings.Prevent_Plan_Requirement_Deletion__c = true; 
-            }
-            upsert orgSettings;
+            orgSettings = enablePreventForAccountsAndPlanRequirement();
         }
+        if (isNewInstallOrPushUpgradedTo196) {
+            UTIL_Debug.debug('***Upgrading to version 1.96 or new installs');
+            orgSettings = enablePreventPPlanDeletionHierarchyCustomSettings();
+        }
+        upsert orgSettings;
+    }
+
+    /*****************************************************************************************************************************************************************
+    * @description Enables Prevent Account Deletion and Plan Requirement Deletion by default on Hierarchy Custom Settings for new installs and when upgrading to 1.95
+    * @return void
+    */
+    global static Hierarchy_Settings__c enablePreventForAccountsAndPlanRequirement() {
```

Comment:
Please split this out into two methods, one for account deletion and one for plan requirement deletion? I'd prefer that we have one method per task so we can pinpoint any errors that might occur.

Please also add unit tests for those methods.